analyzers/tests/SonarAnalyzer.UnitTest/LiveVariableAnalysis/SonarLiveVariableAnalysisTest.cs

```diff
@@ -504,6 +504,62 @@ static int LocalFunction(int a)
             context.Validate(context.Cfg.EntryBlock, new LiveIn("a"), new LiveOut("a"));
         }
 
+        [TestMethod]
+        public void UsingBlock_LiveInUntilTheEnd()
+        {
+            /*     Jump
+             *     new ms
+             *       |
+             *     Binary
+             *     Method(ms.Length)
+             *      /  \
+             * Simple   \
+             * Method(0) |
+             *      \   /
+             *     Using
+             *       |
+             *     Exit
+             */
+            var code = @"
+using (var ms = new System.IO.MemoryStream())
+{
+    Method(ms.Length);
+    if (boolParameter)
+        Method(0);
+}";
+            var context = new Context(code);
+            context.Validate(context.Block<JumpBlock>(), new LiveIn("boolParameter"), new LiveOut("boolParameter", "ms"));
+            context.Validate(context.Block<BinaryBranchBlock>(), new LiveIn("boolParameter", "ms"), new LiveOut("ms"));
+            context.Validate(context.Block<SimpleBlock>(), new LiveIn("ms"), new LiveOut("ms"));
+            context.Validate(context.Block<UsingEndBlock>(), new LiveIn("ms"));
+            context.Validate(context.Block<ExitBlock>());
+        }
+
+        [TestMethod]
+        public void UsingVar_NotSupported()
+        {
+            /* Bad CFG Shape
```

Comment:
I see no difference between this one and Using_LiveInUntilTheEnd. If so, why are the UCFGs in comments differ one from another?

Reply:
The tests are quite different as we're in the old Sonar context.

The old block syntax `using (var ms = new System.IO.MemoryStream()) { .. }` in `Using_LiveInUntilTheEnd` behaves as expected. Contains initial `Jump` block and end `Using` block before the exit. Also `LiveIn` and `LiveOut` makes sense.

The new `using var ms = new System.IO.MemoryStream()` is not supported in CFG => initial `Jump` and the ending `Using` blocks are completely missing in the CFG shape. So the `LiveIn` and `LiveOut` is not properly supported here.